modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/execution/SdkInternalContext.java

```diff
@@ -46,62 +46,63 @@ public static SdkInternalContext from(CoreEvent event) {
    * SDK components may be nested within each other, so some of the context must be kept separately for the component it belongs
    * to.
    */
-  private final Deque<LocationSpecificSdkInternalContext> locationSpecificContext = new ArrayDeque<>();
+  private final Map<ComponentLocation, LocationSpecificSdkInternalContext> locationSpecificContext = new SmallMap<>();
```

Comment:
are you sure that `ComponentLocatin#equals()` is fast enough for `SmallMap` ?

Reply:
it does equals on String, which is cheaper that calculating all of its hashes